Fix #6659: Bus stations can be demolished when not in demolish mode

alex6m59 opened the pull request:
For Bus and Road stations only, if you are in demolish mode and click on the station
without releasing the button. Then you cancel demolish mode with R key.
Finally you release the mouse button. The station was demolished, instead of being built.

The demolish mode was not checked when mouse up event occured.

Commit:
Fix #6659: Bus stations can be demolished when not in demolish mode
For Bus and Road stations only, if you are in demolish mode and click on the station
without releasing the button. Then you cancel demolish mode with R key.
Finally you release the mouse button. The station was demolished, instead of being built.

The demolish mode was not checked when mouse up event occured.

PeterN commented on src/road_gui.cpp:
```diff
@@ -633,22 +633,28 @@ struct BuildRoadToolbarWindow : Window {
 				case DDSP_REMOVE_BUSSTOP: {
-					TileArea ta(start_tile, end_tile);
-					DoCommandP(ta.tile, ta.w | ta.h << 8, (_ctrl_pressed << 1) | ROADSTOP_BUS, CMD_REMOVE_ROAD_STOP | CMD_MSG(_road_type_infos[_cur_roadtype].err_remove_station[ROADSTOP_BUS]), CcPlaySound_SPLAT_OTHER);
+					if(this->IsWidgetLowered(WID_ROT_BUS_STATION)) {
```
Space between if and ( please.

PeterN commented on src/road_gui.cpp:
```diff
@@ -633,22 +633,28 @@ struct BuildRoadToolbarWindow : Window {
-					PlaceRoadStop(start_tile, end_tile, (_ctrl_pressed << 5) | RoadTypeToRoadTypes(_cur_roadtype) << 2 | ROADSTOP_TRUCK, CMD_BUILD_ROAD_STOP | CMD_MSG(_road_type_infos[_cur_roadtype].err_build_station[ROADSTOP_TRUCK]));
-					break;
-
 				case DDSP_REMOVE_BUSSTOP: {
```
Braces are no longer needed for this switch case.

alex6m59 committed:
Fix #6659: Bus stations can be demolished when not in demolish mode
For Bus and Road stations only, if you are in demolish mode and click on the station
without releasing the button. Then you cancel demolish mode with R key.
Finally you release the mouse button. The station was demolished, instead of being built.

The demolish mode was not checked when mouse up event occured.
